lib/heap_sort.rb

```diff
@@ -1,8 +1,18 @@
 
 
 # This method uses a heap to sort an array.
-# Time Complexity:  ?
-# Space Complexity: ?
-def heap_sort(list)
-  raise NotImplementedError, "Method not implemented yet..."
+# Time Complexity:  O(n)
+# Space Complexity: O(n)
+def heapsort(list)
```

Comment:
👍 Time complexity is O(n log n) because you run `mamasHeap.add` (which takes Log n time) n times.  So n * log n = n log n